File: Bridge/Resources/.generated/system/iO/binaryReader.js
    Bridge.define("System.IO.BinaryReader", {
        inherits: [System.IDisposable],
        props: {
            BaseStream: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            }
        },
        alias: ["dispose", "System$IDisposable$dispose"],
        ctors: {
            ctor: function (input) {
                System.IO.BinaryReader.$ctor1.call(this, input, System.Text.Encoding.UTF8UnmarkedUnsafe);
            },
            $ctor1: function (input, encoding) {
                this.$initialize();
            }
        },
        methods: {
            close: function () {
                throw new System.NotImplementedException();
            },
            dispose: function () {
                throw new System.NotImplementedException();
            },
            peekChar: function () {
                throw new System.NotImplementedException();
            },
            read: function () {
                throw new System.NotImplementedException();
            },
            read$1: function (buffer, index, count) {
                throw new System.NotImplementedException();
            },
            read$2: function (buffer, index, count) {
                throw new System.NotImplementedException();
            },
            readBoolean: function () {
                throw new System.NotImplementedException();
            },
            readByte: function () {
                throw new System.NotImplementedException();
            },
            readBytes: function (count) {
                throw new System.NotImplementedException();
            },
            readChar: function () {
                throw new System.NotImplementedException();
            },
            readChars: function (count) {
                throw new System.NotImplementedException();
            },
            readDecimal: function () {
                throw new System.NotImplementedException();
            },
            readDouble: function () {
                throw new System.NotImplementedException();
            },
            readInt16: function () {
                throw new System.NotImplementedException();
            },
            readInt32: function () {
                throw new System.NotImplementedException();
            },
            readInt64: function () {
                throw new System.NotImplementedException();
            },
            readSByte: function () {
                throw new System.NotImplementedException();
            },
            readString: function () {
                throw new System.NotImplementedException();
            },
            readSingle: function () {
                throw new System.NotImplementedException();
            },
            readUInt16: function () {
                throw new System.NotImplementedException();
            },
            readUInt32: function () {
                throw new System.NotImplementedException();
            },
            readUInt64: function () {
                throw new System.NotImplementedException();
            }
        }
    });

File: Bridge/System/IO/BinaryReader.cs
using Bridge;
using System.Text;

namespace System.IO
{
    public class BinaryReader : IDisposable
    {
        public BinaryReader(Stream input)
            : this(input, Encoding.UTF8UnmarkedUnsafe)
        {
        }
        
        public BinaryReader(Stream input, Encoding encoding)
        {
        }

        public virtual Stream BaseStream
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public virtual void Close()
        {
            throw new NotImplementedException();
        }
        
        public void Dispose ()
        {
            throw new NotImplementedException();
        }
        
        public virtual int PeekChar()
        {
            throw new NotImplementedException();
        }

        public virtual int Read()
        {
            throw new NotImplementedException();
        }

        public virtual int Read(byte[] buffer, int index, int count)
        {
            throw new NotImplementedException();
        }

        public virtual int Read(char[] buffer, int index, int count)
        {
            throw new NotImplementedException();
        }

        public virtual bool ReadBoolean()
        {
            throw new NotImplementedException();
        }

        public virtual byte ReadByte()
        {
            throw new NotImplementedException();
        }

        public virtual byte[] ReadBytes(int count)
        {
            throw new NotImplementedException();
        }

        public virtual char ReadChar()
        {
            throw new NotImplementedException();
        }

        public virtual char[] ReadChars(int count)
        {
            throw new NotImplementedException();
        }

        unsafe public virtual decimal ReadDecimal()
        {
            throw new NotImplementedException();
        }

        public virtual double ReadDouble()
        {
            throw new NotImplementedException();
        }

        public virtual short ReadInt16()
        {
            throw new NotImplementedException();
        }

        public virtual int ReadInt32()
        {
            throw new NotImplementedException();
        }

        public virtual long ReadInt64()
        {
            throw new NotImplementedException();
        }

        public virtual sbyte ReadSByte()
        {
            throw new NotImplementedException();
        }

        public virtual string ReadString()
        {
            throw new NotImplementedException();
        }

        public virtual float ReadSingle()
        {
            throw new NotImplementedException();
        }

        public virtual ushort ReadUInt16()
        {
            throw new NotImplementedException();
        }

        public virtual uint ReadUInt32()
        {
            throw new NotImplementedException();
        }

        public virtual ulong ReadUInt64()
        {
            throw new NotImplementedException();
        }
    }
}